lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -346,81 +372,65 @@ def main():
-
-        elif state == "import":
-            if module.check_mode:
-                module.exit_json(changed=True, db=db_name)
-            rc, stdout, stderr = db_import(module, login_host, login_user,
-                                           login_password, db, target,
-                                           all_databases,
-                                           login_port, config_file,
-                                           socket, ssl_cert, ssl_key, ssl_ca)
-            if rc != 0:
-                module.fail_json(msg="%s" % stderr)
-            else:
-                module.exit_json(changed=True, db=db_name, msg=stdout)
 
-        elif state == "present":
-            module.exit_json(changed=False, db=db_name)
+    existence_list = []
+    non_existence_list = []
 
-    else:
-        if state == "present":
-            if module.check_mode:
-                changed = True
-            else:
-                try:
-                    changed = db_create(cursor, db, encoding, collation)
-                except Exception as e:
-                    module.fail_json(msg="error creating database: %s" % to_native(e),
-                                     exception=traceback.format_exc())
-            module.exit_json(changed=changed, db=db_name)
-
-        elif state == "import":
-            if module.check_mode:
-                module.exit_json(changed=True, db=db_name)
+    if not all_databases:
+        for each_database in db:
+            if db_exists(cursor, [each_database]):
+                existence_list.append(each_database)
             else:
-                try:
-                    changed = db_create(cursor, db, encoding, collation)
-                    if changed:
-                        rc, stdout, stderr = db_import(module, login_host, login_user,
-                                                       login_password, db, target, all_databases,
-                                                       login_port, config_file, socket, ssl_cert, ssl_key, ssl_ca)
-                        if rc != 0:
-                            module.fail_json(msg="%s" % stderr)
-                        else:
-                            module.exit_json(changed=True, db=db_name, msg=stdout)
-                except Exception as e:
-                    module.fail_json(msg="error creating database: %s" % to_native(e),
-                                     exception=traceback.format_exc())
-
-        elif state == "absent":
-            module.exit_json(changed=False, db=db_name)
-
-        elif state == "dump":
-            if module.check_mode:
-                module.exit_json(changed=False, db=db_name)
-            module.fail_json(msg="Cannot dump database %r - not found" % (db_name))
+                non_existence_list.append(each_database)
+
+    if state == "absent":
+        if module.check_mode:
+            module.exit_json(changed=bool(existence_list), db=db_name, db_list=db)
```

Comment:
Their content is same however, the `type` is different.

I kept `db` and `db_list` since I didn't want to break existing playbooks of users which explicitly checks the content of `result.db` as [here](https://app.shippable.com/github/ansible/ansible/runs/132137/55/console) eventually breaking their playbook.
```
01:19 fatal: [testhost]: FAILED! => {
01:19     "assertion": "result.db =='data'",
01:19     "changed": false,
01:19     "evaluated_to": false,
01:19     "msg": "Assertion failed"
01:19 }
```

I need your opinion on this to know if I should be keeping both the variables or should I just be returning list of databases
```
-        module.exit_json(changed=changed, db=db_name, db_list=db)
+        module.exit_json(changed=changed, db=db)
```

Reply:
I think keeping both is probably best. Changing types of an existing return value, or removing one, will definitely break some people's playbooks.

You could deprecate the old return value and announce that in the porting guide; then the old return value can be dropped in 2.13. This is not optimal because there is no mechanism which tells people that the result they are using is deprecated, but at least people who read the porting guide and/or changelog will have four versions to adapt their playbooks.

Reply:
@felixfontein , I thought that it was added by this PR, if it was there before, sure, it must be kept !
@pratikgadiya12 , dont remove it, please! :)

Reply:
@Andersson007 I interpreted @pratikgadiya12's statement this way. Now I also checked, it was there before, indeed. What's completely missing (and has been in the past) is documentation for the return values! @pratikgadiya12 can you add this? I guess `db` has been there all the time, and `db_list` will be `version_added: 2.9`.

Reply:
Ah, I just saw you already mentioned the return value docs :)

Reply:
Will keep both the return values and add the return value docs